x-pack/test/osquery_cypress/runner.ts

```diff
@@ -48,12 +49,12 @@ async function setupFleetAgent({ getService }: FtrProviderContext) {
   const policyEnrollmentKey = await createAgentPolicy(kbnClient, log, 'Default policy');
   const policyEnrollmentKeyTwo = await createAgentPolicy(kbnClient, log, 'Osquery policy');
 
-  await new AgentManager(policyEnrollmentKey, config.get('servers.fleetserver.port'), log).setup();
-  await new AgentManager(
-    policyEnrollmentKeyTwo,
-    config.get('servers.fleetserver.port'),
-    log
-  ).setup();
+  const port = config.get('servers.fleetserver.port');
+
+  const agent = await new AgentManager(policyEnrollmentKey, port, log).setup();
+  await waitForHostToEnroll(kbnClient, agent);
```

Comment:
maybe we can move `waitForHostToEnroll` into `AgentManager`?

Reply:
I think we can, I had a different approach in place for firstly waitForHostToEnroll and if failed to create a new agent, but I do not do it now so this can be easily moved to agentManager if you think it makes sense